packages/pds/src/app-view/services/graph/index.ts

```diff
@@ -0,0 +1,126 @@
+import Database from '../../../db'
+import { DbRef } from '../../../db/util'
+import { NotEmptyArray } from '@atproto/common'
+import { sql } from 'kysely'
+import { ImageUriBuilder } from '../../../image/uri'
+import { ProfileView } from '../../../lexicon/types/app/bsky/actor/defs'
+import { List } from '../../db/tables/list'
+
+export class GraphService {
+  constructor(public db: Database, public imgUriBuilder: ImageUriBuilder) {}
+
+  static creator(imgUriBuilder: ImageUriBuilder) {
+    return (db: Database) => new GraphService(db, imgUriBuilder)
+  }
+
+  getListsQb(requester: string) {
+    const { ref } = this.db.db.dynamic
+    return this.db.db
+      .selectFrom('list')
+      .innerJoin('did_handle', 'did_handle.did', 'list.creator')
+      .selectAll('list')
+      .selectAll('did_handle')
+      .select(
+        this.db.db
+          .selectFrom('list_mute')
+          .where('list_mute.mutedByDid', '=', requester)
+          .whereRef('list_mute.listUri', '=', ref('list.uri'))
+          .select('list_mute.listUri')
+          .as('viewerMuted'),
+      )
+  }
+
+  getListItemsQb() {
+    return this.db.db
+      .selectFrom('list_item')
+      .innerJoin('did_handle as subject', 'subject.did', 'list_item.subjectDid')
+      .selectAll('subject')
+      .select([
+        'list_item.cid as cid',
+        'list_item.createdAt as createdAt',
+        'list_item.reason as reason',
+        'list_item.reasonFacets as reasonFacets',
+      ])
+  }
+
+  blockQb(requester: string, refs: NotEmptyArray<DbRef>) {
+    const subjectRefs = sql.join(refs)
+    return this.db.db
+      .selectFrom('actor_block')
+      .where((qb) =>
+        qb
+          .where('actor_block.creator', '=', requester)
+          .whereRef('actor_block.subjectDid', 'in', sql`(${subjectRefs})`),
+      )
+      .orWhere((qb) =>
+        qb
+          .where('actor_block.subjectDid', '=', requester)
+          .whereRef('actor_block.creator', 'in', sql`(${subjectRefs})`),
+      )
```

Comment:
These "or"s can create sneaky problems.  If there's a chance a consumer might add additional conditions to the where clause, it could be worth adding some parens around these two conditions.  If that's unlikely, then this should be fine 😎

Reply:
yeah that's a good call. those are sneaky 👌